src/views/workspaceNode.ts

```diff
@@ -18,7 +19,9 @@ export class WorkspaceNode extends DataNode {
         if (!this.nodeData.uri) {
             return undefined;
         }
-        return Jdtls.getProjects(this.nodeData.uri);
+        const result = await Jdtls.getProjects(this.nodeData.uri);
+        PromotionProvider.getInstance().checkJavaVersion(result);
+        return result;
```

Comment:
1. what if this method is called >1 times?
2. The getData function should only handle data retrieval. Performing a Java version validation at the same time as data fetching seems quite unreasonable. This is clearly a side effect and should be avoided.